tensorly/tenalg/proximal.py

```diff
@@ -91,4 +91,316 @@ def procrustes(matrix):
+    UtM: r-by-n array
+        Pre-computed product of the transposed of U and M, used in the update rule
+    UtU: r-by-r array
+        Pre-computed product of the transposed of U and U, used in the update rule
+    in_V: r-by-n initialization matrix (mutable)
+        Initialized V array
+        By default, is initialized with one non-zero entry per column
+        corresponding to the closest column of U of the corresponding column of M.
+    maxiter: Postivie integer
+        Upper bound on the number of iterations
+        Default: 500
+    delta : float in [0,1]
+        early stop criterion, while err_k > delta*err_0. Set small for
+        almost exact nnls solution, or larger (e.g. 1e-2) for inner loops
+        of a PARAFAC computation.
+        Default: 10e-8
+    sparsity_coefficient: float or None
+        The coefficient controling the sparisty level in the objective function.
+        If set to None, the problem is solved unconstrained.
+        Default: None
+    nonzero_row: boolean
+        True if the lines of the V matrix can't be zero,
+        False if they can be zero
+        Default: False
+
+    Returns
+    -------
+    V: array
+        a r-by-n nonnegative matrix \approx argmin_{V >= 0} ||M-UV||_F^2
+    eps: float
+        number of loops authorized by the error stop criterion
+    cnt: integer
+        final number of update iteration performed
+    rho: float
+        number of loops authorized by the time stop criterion
+
+    References
+    ----------
+    [1]: N. Gillis and F. Glineur, Accelerated Multiplicative Updates and
+    Hierarchical ALS Algorithms for Nonnegative Matrix Factorization,
+    Neural Computation 24 (4): 1085-1105, 2012.
+
+    [2] J. Eggert, and E. Korner. "Sparse coding and NMF."
+    2004 IEEE International Joint Conference on Neural Networks
+    (IEEE Cat. No. 04CH37541). Vol. 4. IEEE, 2004.
+
+    """
+
+    r, n = tl.shape(UtM)
+    if in_V is None:  # checks if V is empty
+        V = tl.solve(UtU, UtM)
+
+        V[V < 0] = 0
+        # Scaling
+        scale = tl.sum(UtM * V) / tl.sum(
+            UtU * tl.dot(V, tl.transpose(V)))
+        V = tl.dot(scale, V)
+    else:
+        V = in_V
+
+    rho = 1
+    eps0 = 0
+    cnt = 1
+    eps = 1
+
+
+    while eps >= delta * eps0 and cnt <= 1 + 0.5* rho and cnt <= maxiter:
+        nodelta = 0
+        for k in range(r):
+
+            if UtU[k, k] != 0:
+                if sparsity_coefficient != None: # Modifying the objective function for sparsification
+                    if tl.get_backend() == 'pytorch':
+                        import torch
+                        deltaV = torch.maximum((UtM[k, :] - UtU[k, :] @ V - sparsity_coefficient * tl.ones(n)) / UtU[k, k],
+                                        -V[k, :])
+                    else:
+                        deltaV = tl.max([(UtM[k, :] - UtU[k, :] @ V - sparsity_coefficient * tl.ones(n)) / UtU[k, k],
+                                        -V[k, :]],axis=0)
```

Comment:
We could probably update the backend to include both `max` and `maximum`, following numpy's convention that max does a reduce while maximum is elementwise.

Reply:
Here, I have a problem with pytorch. Yes, if tensorly includes elementwise for each backend, problem will be solved.

Reply:
Actually @JeanKossaifi we have no experience with the backend system, what would it take to support both max and maximum?

Reply:
It should just be a matter of adding them in each backend, that shouldn't be an issue, I'll have a look. 
On this line, @ is not supported by all backends, it's (more verbose but) better to use `tl.dot()`

Are you using tl.max over a list of two elements? I wonder if just using Python's builtin `max` would work. 
I also don't think the axis is needed here since it's a list withh just two elements. 
Finally, I also don't think you need the tl.ones, you can directly do array - int and the int is broadcast to the whole array.

Reply:
I removed tl.ones and @. 

I am using tl.max over two lists. The target is to find elementwise maximum for each location in kth row. So, the result is also a list which has `len(V(k,:))` elements.